### ISSUE TITLE
Replace discontinued C++ standard library documentation

### ISSUE THREAD

**Issue by lerks** [cms-dev/cms#875]
[NEED FEEDBACK FROM @bblackham @gollux @wil93]

CMS provides contestants access to the documentation of the C++ standard library through CWS. This has always been the SGI (Silicon Graphics) STL doc (http://www.sgi.com/tech/stl/), as it was the one that was used at the IOI. If you actually clicked on the previous link you realized that it doesn't lead anymore to where it used to. Hewlett Packard acquired SGI and shut down the doc. The link now leads to this announcement, posted on Jan 2, 2018: https://community.hpe.com/t5/Servers-The-Right-Compute/SGI-com-Tech-Archive-Resources-now-retired/ba-p/6992583

The package that provided that documentation for offline use in Ubuntu is still available, and will still be at least in 18.04: https://packages.ubuntu.com/bionic/stl-manual
However the corresponding package for Arch Linux has been removed. The same may hold for other distributions. This is making it harder to get the full "CMS experience" on anything other than Ubuntu.

I thus want to propose to move away from that doc. A good replacement seems to be this: http://en.cppreference.com/w/
It is available offline in at least Ubuntu (https://packages.ubuntu.com/bionic/cppreference-doc-en-html) and Arch LInux (https://aur.archlinux.org/packages/cppreference/). It also seems more up-to-date (it includes the latest C++ standards) and easier to navigate.

The other alternatives I found were either outdated and semi-abandoned as well (for example STLport, which is a fork of the SGI STL), bound to a specific vendor (GNU, clang, MSDN, ...) or not available offline (http://www.cplusplus.com/).

I would like the IOI ITC to give their opinion on this, as the IOI may have some constraints or requirements on what documentation is available to contestants.

**Comment by wil93** [comment]
I'm not aware of any regulation about this, but FWIW in the Italian selection rounds we routinely replace that documentation with the one you suggested.

**Comment by andreyv** [comment]
I agree that cppreference.com is the de facto C/C++ reference now. We use it in our contests to replace the STL documentation mentioned above, and in fact we use the same Ubuntu package.

**Comment by gollux** [comment]
I agree that cppreference is simply *the* C++ reference these years and the
ancient SGI STL manual should be replaced by it.

However, I am not so sure that it should be the job of the CMS to provide any
such documentation. Different contests need different documentation, some
include FPC documentation besides C++, some prefer it in a local language, and
so on.

I would prefer to have a configurable set of links to documentation in the CWS
and to provide the actual documentation outside the CWS. Most people already
use Nginx in front of the CWS and it is easy to configure it to serve static
files in some subtree.

**Comment by wil93** [comment]
> However, I am not so sure that it should be the job of the CMS to provide any such documentation

I just remembered that in at least one occasion we provided offline documentation with a software (easily installable with `apt-get` but I can't remember the name).

I think it was during one of our national finals, when we decided that replacing the SGI STL manual was too hard so we disabled it entirely and installed that software.

From a quick search I found https://zealdocs.org/ but it's not the same we used that time.
